pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -475,6 +530,86 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(clusterName string, apiService *api.Ser
 		status.Ingress = append(status.Ingress, api.LoadBalancerIngress{IP: floatIP.FloatingIP})
 	}
 
+	if lbaas.opts.ManageSecurityGroups {
+
+		lbSecGroupCreateOpts := groups.CreateOpts{
+			Name:        getSecurityGroupName(clusterName, apiService),
+			Description: fmt.Sprintf("Securty Group for %v Service LoadBalancer", apiService.Name),
+		}
+
+		lbSecGroup, err := groups.Create(lbaas.network, lbSecGroupCreateOpts).Extract()
+
+		if err != nil {
+			return nil, err
+		}
+
+		for _, port := range ports {
+
+			for _, sourceRange := range sourceRanges.StringSlice() {
+				lbSecGroupRuleCreateOpts := rules.CreateOpts{
+					Direction:      "ingress",
+					PortRangeMax:   int(port.Port),
+					PortRangeMin:   int(port.Port),
+					Protocol:       strings.ToLower(string(port.Protocol)),
+					RemoteIPPrefix: string(sourceRange),
+					SecGroupID:     lbSecGroup.ID,
+					EtherType:      "IPv4",
```

Comment:
This of course assumes `sourceRange` is IPv4.  I suggest:
    ethertype := "IPv4"
    if sourceRange.IP.To4() == nil {
        ethertype = "IPv6"
    }

That's not great either, since we make no effort to check the address family of floating IP and loadbalancer VIP above, but it's a start..